stubs/psycopg2/psycopg2/_psycopg.pyi

```diff
@@ -157,27 +157,27 @@ class Column:
     def __setstate__(self, state): ...
 
 class ConnectionInfo:
-    backend_pid: Any
-    dbname: Any
-    dsn_parameters: Any
-    error_message: Any
-    host: Any
-    needs_password: Any
-    options: Any
-    password: Any
-    port: Any
-    protocol_version: Any
-    server_version: Any
-    socket: Any
-    ssl_attribute_names: Any
-    ssl_in_use: Any
-    status: Any
-    transaction_status: Any
-    used_password: Any
-    user: Any
+    backend_pid: int
+    dbname: str
+    dsn_parameters: dict[str, str]
+    error_message: str | None
+    host: str
+    needs_password: bool
+    options: str
+    password: str
+    port: int
+    protocol_version: int
+    server_version: int
+    socket: int
+    ssl_attribute_names: list[str]
+    ssl_in_use: bool
+    status: int
+    transaction_status: int
+    used_password: bool
+    user: str
```

Comment:
On this and the other parameters possibly being None: it looks like None is returned only when the corresponding libpq functions return NULL. I haven't done a thorough audit, but from a quick scan of https://github.com/postgres/postgres/blob/master/src/interfaces/libpq/fe-connect.c#L6754 it looks like libpq only returns NULL if you give it a NULL ptr in the first place; in some places they explicitly choose to return an empty string rather than NULL.

Given that, plus the fact that psycopg2's docs never mention `None` for these types, I think I'd be inclined to leave these types as they are. Does that sound reasonable?

Reply:
Sounds good, but could you add a comment to the code pointing out that while technically they can be None, that's highly unlikely in practice?

Reply:
Certainly, see ed4d8df3